src/components/CategoryFilter/CategoryFilter.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { useMemo } from 'react'
+import classNames from 'classnames'
+
+import { Item, Props } from './CategoryFilter.types'
+
+import './CategoryFilter.css'
+
+export const CategoryFilter = ({ title, items, value, onClick }: Props) => {
+  const branch = useMemo(() => {
+    const branch = new Set<string>()
+
+    const buildBranch = (items: Item[]) => {
```

Comment:
What do you think about renaming the name of this method to:
```suggestion
    const buildBranchOfSelectedItem = (items: Item[]) => {
```

Reply:
Commited the change without using suggestion because it had to be changed in other places as well